test/upgrade/2.0.0-2.1.0.js

```diff
@@ -175,16 +194,561 @@ describe("[@skip-on-coverage] After facet upgrade, everything is still operation
+      });
+
+      it("Seller can be updated in two steps", async function () {
+        const oldSeller = sellers[0][3];
+
+        const seller = oldSeller.seller.clone();
+        seller.treasury = treasury.address;
+        // Treasury is the only values that can be update without address owner authorization
+        let sellerStruct = seller.toStruct();
+
+        seller.admin = admin.address;
+        seller.operator = operator.address;
+        seller.clerk = clerk.address;
+
+        const pendingSellerUpdate = seller.clone();
+        pendingSellerUpdate.id = "0";
+        pendingSellerUpdate.treasury = ethers.constants.AddressZero;
+        pendingSellerUpdate.active = false;
+        let pendingSellerUpdateStruct = pendingSellerUpdate.toStruct();
+
+        const authToken = mockAuthToken();
+        const pendingAuthToken = authToken.clone();
+        const oldSellerAuthToken = oldSeller.authToken.toStruct();
+        const pendingAuthTokenStruct = pendingAuthToken.toStruct();
+
+        // Update seller
+        let tx = await accountHandler.connect(oldSeller.wallet).updateSeller(seller, authToken);
+
+        // Testing for the SellerUpdateApplied event
+        await expect(tx)
+          .to.emit(accountHandler, "SellerUpdateApplied")
+          .withArgs(
+            seller.id,
+            sellerStruct,
+            pendingSellerUpdateStruct,
+            oldSellerAuthToken,
+            pendingAuthTokenStruct,
+            oldSeller.wallet.address
+          );
+
+        // Testing for the SellerUpdatePending event
+        await expect(tx)
+          .to.emit(accountHandler, "SellerUpdatePending")
+          .withArgs(seller.id, pendingSellerUpdateStruct, pendingAuthTokenStruct, oldSeller.wallet.address);
+
+        // Update seller operator
+        tx = await accountHandler.connect(operator).optInToSellerUpdate(seller.id, [SellerUpdateFields.Operator]);
+
+        pendingSellerUpdate.operator = ethers.constants.AddressZero;
+        pendingSellerUpdateStruct = pendingSellerUpdate.toStruct();
+        seller.clerk = oldSeller.seller.clerk;
+        seller.admin = oldSeller.seller.admin;
+        sellerStruct = seller.toStruct();
+
+        // Check operator update
+        await expect(tx)
+          .to.emit(accountHandler, "SellerUpdateApplied")
+          .withArgs(
+            seller.id,
+            sellerStruct,
+            pendingSellerUpdateStruct,
+            oldSellerAuthToken,
+            pendingAuthTokenStruct,
+            operator.address
+          );
+
+        // Update seller clerk
+        tx = await accountHandler.connect(clerk).optInToSellerUpdate(seller.id, [SellerUpdateFields.Clerk]);
+
+        pendingSellerUpdate.clerk = ethers.constants.AddressZero;
+        pendingSellerUpdateStruct = pendingSellerUpdate.toStruct();
+        seller.clerk = clerk.address;
+        seller.admin = oldSeller.seller.admin;
+        sellerStruct = seller.toStruct();
```

Comment:
The operator update was just confirmed above. shouldn't seller.operator = seller.address at this point?

Reply:
I changed the code to be more cleare what's going on.